### ISSUE TITLE
question about dilation

### ISSUE THREAD

**Issue by junhyk** [locuslab/trellisnet#5]
Hi,

I have a question about the implementation of dilation in `class TrellisNet`
In the code trellisnet.py
```
# Feed-forward layers
for i in range(0, self.nlevels):
    d = dilation_cycle[i % len(dilation_cycle)]
    Z = self.step(Z, dilation=d, hc=hc)
    if aux and i % self.aux_frequency == (self.aux_frequency-1):
        aux_outs.append(Z[:, -nout:].unsqueeze(0))
```
I found you cycling the dilation parameter.
What is the reason behind the cycling?

Thanks

**Comment by shaojieb** [comment]
Hi @zzoon91,

It's mainly a choice of design. As TrellisNets can be deep (e.g., we use >100 layers in char-LM tasks), using an exponential expansion of the dilation size would be both unreasonable and computationally inefficient. Therefore, we adopted a scheme similar to the one used in WaveNet; i.e., cycling the dilation sizes. But generally, this is not necessary. For example, in pixel MNIST, we only actually used one cycle.

Hope that helps!
